materialize-databricks/driver.go

```diff
@@ -477,11 +514,43 @@ func (d *transactor) Acknowledge(ctx context.Context) (*pf.ConnectorState, error
 				}
 				return nil, fmt.Errorf("query %q failed: %w", query, err)
 			}
+
+		}
+
+		var binding = d.bindingForStateKey(stateKey)
```

Comment:
You might run this both before and after the Store query is run, that way we can tell if there were already duplicates in the table before our query ran.